tests/influence/test_influences.py

```diff
@@ -4,57 +4,105 @@
+    x: NDArray[np.float_],
+    y: NDArray[np.float_],
+    x_test: NDArray[np.float_],
+    y_test: NDArray[np.float_],
+    influence_type: InfluenceType = InfluenceType.Up,
+    hessian_regularization: float = 0,
+):
+    """Calculates analytically the influence of each training sample on the
+     test samples for an ordinary least squares model (Ax+b=y with quadratic
+     loss).
+
+    :param linear_model: A tuple of arrays of shapes (N, M) and N representing A
+        and b respectively.
+    :param x: An array of shape (M, K) containing the features of the
+        training set.
+    :param y: An array of shape (M, L) containing the targets of the
+        training set.
+    :param x_test: An array of shape (N, K) containing the features of the test
+        set.
+    :param y_test: An array of shape (N, L) containing the targets of the test
+        set.
+    :param influence_type: the type of the influence.
+    :param hessian_retularization: regularization value for the hessian
+    :returns: An array of shape (B, C) with the influences of the training points
+        on the test points if influence_type is "up", an array of shape (K, L,
+        M) if influence_type is "perturbation".
+    """
+
+    s_test_analytical = linear_analytical_influence_factors(
+        linear_model, x, y, x_test, y_test, hessian_regularization
     )
-    from pydvl.influence.model_wrappers import TorchLinearRegression, TorchMLP
-    from pydvl.utils.dataset import load_wine_dataset
-except ImportError:
-    pass
-
+    if influence_type == InfluenceType.Up:
+        train_grads_analytical = linear_derivative_analytical(
+            linear_model,
+            x,
+            y,
+        )
+        result: NDArray = np.einsum(
+            "ia,ja->ij", s_test_analytical, train_grads_analytical
+        )
+    elif influence_type == InfluenceType.Perturbation:
+        train_second_deriv_analytical = linear_mixed_second_derivative_analytical(
+            linear_model,
+            x,
+            y,
+        )
+        result: NDArray = np.einsum(
+            "ia,jab->ijb", s_test_analytical, train_second_deriv_analytical
+        )
+    return result
 
-class InfluenceTestSettings:
-    DATA_OUTPUT_NOISE: float = 0.01
-    ACCEPTABLE_ABS_TOL_INFLUENCE: float = 5e-4
-    ACCEPTABLE_ABS_TOL_INFLUENCE_CG: float = 1e-3
 
-    INFLUENCE_TEST_CONDITION_NUMBERS: List[int] = [5]
-    INFLUENCE_TRAINING_SET_SIZE: List[int] = [500]
-    INFLUENCE_TEST_SET_SIZE: List[int] = [20]
-    INFLUENCE_N_JOBS: List[int] = [1]
-    INFLUENCE_DIMENSIONS: List[Tuple[int, int]] = [
-        (10, 10),
-        (20, 10),
-        (3, 20),
-        (20, 20),
-    ]
+INFLUENCE_TEST_CONDITION_NUMBERS: List[int] = [3]
+INFLUENCE_TRAINING_SET_SIZE: List[int] = [50, 30]
+INFLUENCE_TEST_SET_SIZE: List[int] = [20]
+INFLUENCE_DIMENSIONS: List[Tuple[int, int]] = [
+    (10, 10),
+    (3, 20),
+]
+HESSIAN_REGULARIZATION: List[float] = [0, 1]
 
 
 test_cases = list(
```

Comment:
As mentioned in #306, this seems like reinventing `@pytest.mark.parametrize`. Just stack calls to this decorator to get a product.